setup.py

```diff
@@ -1,231 +1,261 @@
-        mod_path = mod_name.replace('.', '/') + '.pyx'
-        e = Extension(mod_name, [mod_path], language=language, include_dirs=includes)
-        exts.append(e)
-    distutils.core.setup(
-        name='spacy',
-        packages=['spacy', 'spacy.tokens', 'spacy.en', 'spacy.serialize',
-                  'spacy.syntax', 'spacy.munge'],
-        description="Industrial-strength NLP",
-        author='Matthew Honnibal',
-        author_email='honnibal@gmail.com',
-        version=VERSION,
-        url="http://spacy.io",
-        package_data=PACKAGE_DATA,
-        ext_modules=exts,
-        cmdclass={'build_ext': build_ext_cython_subclass},
-        license="MIT",
-    )
+    include_dirs = [
+        get_python_inc(plat_specific=True),
+        os.path.join(src_path, 'include')]
 
+    ext_modules = []
+    for mod_name in MOD_NAMES:
+        mod_path = mod_name.replace('.', '/') + '.cpp'
+        ext_modules.append(
+            Extension(mod_name, [mod_path],
+                language='c++', include_dirs=include_dirs))
 
-def run_setup(exts):
-    setup(
+    metadata = dict(
         name='spacy',
-        packages=['spacy', 'spacy.tokens', 'spacy.en', 'spacy.serialize',
-                  'spacy.syntax', 'spacy.munge',
-                  'spacy.tests',
-                  'spacy.tests.matcher',
-                  'spacy.tests.morphology',
-                  'spacy.tests.munge',
-                  'spacy.tests.parser',
-                  'spacy.tests.serialize',
-                  'spacy.tests.spans',
-                  'spacy.tests.tagger',
-                  'spacy.tests.tokenizer',
-                  'spacy.tests.tokens',
-                  'spacy.tests.vectors',
-                  'spacy.tests.vocab'],
-        description="Industrial-strength NLP",
+        packages=PACKAGES,
+        description='Industrial-strength NLP',
         author='Matthew Honnibal',
-        author_email='honnibal@gmail.com',
+        author_email='matt@spacy.io',
         version=VERSION,
-        url="http://honnibal.github.io/spaCy/",
-        package_data=PACKAGE_DATA,
-        ext_modules=exts,
-        license="MIT",
+        url='https://spacy.io',
+        license='MIT',
+        ext_modules=ext_modules,
         install_requires=['numpy', 'murmurhash == 0.24', 'cymem == 1.30', 'preshed == 0.44',
-                          'thinc == 4.0.0', "text_unidecode", 'plac', 'six',
-                          'ujson', 'cloudpickle', 'sputnik == 0.5.2'],
-        setup_requires=["headers_workaround"],
-        cmdclass = {'build_ext': build_ext_subclass },
+                          'thinc == 4.0.0', 'text_unidecode', 'plac', 'six',
+                          'ujson', 'cloudpickle', 'sputnik == 0.6.2'],
+        cmdclass = {
+            'build_ext': build_ext_subclass},
     )
 
-    import headers_workaround
+    # Run build
+    cwd = os.path.abspath(os.path.dirname(__file__))
+    if not os.path.exists(os.path.join(cwd, 'PKG-INFO')):
```

Comment:
Is this conditional detecting whether we're building the package vs. installing it (via pip, conda etc)?

If so, is the check we're using brittle?

We should also think about whether we want to use `__file__`. I think there's an API for this, I think it's called `packageinfo` or something.

If checking for build vs install is what we're doing, and this implementation is the best approach, I'd suggest moving the check to a function named liked `is_building` or whatever, so that it's a little bit more obvious how to change it in future, if we have to update it for new versions of setuptools.